File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/GoBildaRgbLight.java
package org.firstinspires.ftc.teamcode.robot;

import androidx.annotation.NonNull;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;

/** goBILDA RGB LED light on a servo channel. Wires {@link org.marsroboticsassociation.controllib.hardware.GoBildaRgbLight} to hardware. */
public class GoBildaRgbLight extends org.marsroboticsassociation.controllib.hardware.GoBildaRgbLight {

    public GoBildaRgbLight(@NonNull HardwareMap hardwareMap) {
        super(hardwareMap.get(Servo.class, "rgb")::setPosition);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/GoBildaRgbLight.kt
package org.firstinspires.ftc.teamcode.robot

import com.qualcomm.robotcore.hardware.HardwareMap
import com.qualcomm.robotcore.hardware.Servo
import org.marsroboticsassociation.controllib.hardware.GoBildaRgbLight as ControlLibRgbLight

/**
 * Wires ControlLib's goBILDA RGB LED to the `rgb` servo. The library class is `final`, so this is a
 * factory rather than a subclass.
 */
fun GoBildaRgbLight(hardwareMap: HardwareMap): ControlLibRgbLight =
    ControlLibRgbLight(hardwareMap.get(Servo::class.java, "rgb")::setPosition)
